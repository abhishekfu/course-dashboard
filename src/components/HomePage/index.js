import React from 'react';
import {Grid,Container, Typography} from '@material-ui/core'
import { withStyles } from '@material-ui/core/styles';
import styles from './style';
import UserInfo from '../UserInfo';
import AggrerateDisplay from '../AggrerateDisplay';
import IndividualDisplay from '../IndividualDisplay';
function HomePage({classes}) {
    return (
        <Container maxWidth='lg'>
        <Grid container direction='column'>
            <UserInfo/>
            <Typography data-testid='heading' className={classes.text}>Dashboard</Typography>
            <Typography data-testid='subheading' className={classes.subtext}>Mobile UX/UI Design course</Typography>
            <AggrerateDisplay/>
            <IndividualDisplay/>
        </Grid>
        </Container>
    )
}
export default withStyles(styles)(HomePage);