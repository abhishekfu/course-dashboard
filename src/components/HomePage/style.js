const styles = (theme) => ({
    text:{
        fontSize:'30px',
        lineHeight:'36px',
        letterSpacing:'1%',
        fontWeight:'600',
        textTransform:'capitalize',
        fontFamily:'Gilroy'
    },
    subtext:{
        fontSize:'14px',
        fontWeight:'500',
        lineHeight:'16px',
        letterSpacing:'1%',
        textTransform:'capitalize',
        color: '#A6ACBE',
        fontFamily:'Gilroy'
    }
})

export default styles;