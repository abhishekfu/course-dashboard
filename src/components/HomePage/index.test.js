import { render,cleanup } from '@testing-library/react';
import HomePage from './index';

afterEach(cleanup);
it('should take a snapshot', () => {
  const { asFragment } = render(<HomePage />);
  expect(asFragment(<HomePage />)).toMatchSnapshot();
});

it('should equal to "Dashboard"', () => {
    const { getByTestId } = render(<HomePage />); 
    expect(getByTestId('heading')).toHaveTextContent('Dashboard')
});

it('should equal to "Mobile UX/UI Design course"', () => {
    const { getByTestId } = render(<HomePage />); 
    expect(getByTestId('subheading')).toHaveTextContent('Mobile UX/UI Design course')
});