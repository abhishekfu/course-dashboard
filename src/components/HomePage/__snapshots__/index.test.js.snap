// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        class="MuiGrid-root UserInfo-root-5 MuiGrid-item"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row-reverse"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
                >
                  <p
                    class="MuiTypography-root UserInfo-text-3 MuiTypography-body1 MuiTypography-alignRight"
                    data-testid="username"
                  >
                    cody simmons
                  </p>
                  <p
                    class="MuiTypography-root UserInfo-subtext-4 MuiTypography-body1 MuiTypography-alignRight"
                    data-testid="designation"
                  >
                    lecturer
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                  />
                </svg>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root UserInfo-icon-6 MuiSvgIcon-fontSizeLarge"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <p
        class="MuiTypography-root HomePage-text-1 MuiTypography-body1"
        data-testid="heading"
      >
        Dashboard
      </p>
      <p
        class="MuiTypography-root HomePage-subtext-2 MuiTypography-body1"
        data-testid="subheading"
      >
        Mobile UX/UI Design course
      </p>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          data-testid="loading"
        >
          Loading...
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiGrid-root IndividualDisplay-container-8 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
              >
                <p
                  class="MuiTypography-root IndividualDisplay-text-9 MuiTypography-body1"
                >
                  Students by average mark
                </p>
                <div
                  class="MuiFormControl-root IndividualDisplay-input-10"
                >
                  <div
                    class="MuiInputBase-root MuiInput-root IndividualDisplay-input-10 MuiInputBase-formControl MuiInput-formControl"
                  >
                    <div
                      aria-haspopup="listbox"
                      class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                      role="button"
                      tabindex="0"
                    >
                      Descending
                    </div>
                    <input
                      aria-hidden="true"
                      class="MuiSelect-nativeInput"
                      tabindex="-1"
                      value="desc"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSelect-icon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="MuiDivider-root"
            />
            <div
              class="MuiGrid-root IndividualDisplay-list-11 MuiGrid-item"
            >
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
